pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -451,7 +451,16 @@ func (f *ring0Factory) Resumer(info *resource.Info) ([]byte, error) {
 }
 
 func (f *ring0Factory) DefaultNamespace() (string, bool, error) {
-	return f.clientConfig.Namespace()
+	namespace, boolRetval, err := f.clientConfig.Namespace()
```

Comment:
no, it really has to be handled deeper... in the case where they specify a "valid" value of `namespaces/foo`, `clientConfig.Namespace()` would return a value that would be invalid to use directly as a namespace.